### DISCUSSION TITLE
shape.finish() somtimes drawing dashed lines

### DISCUSSION THREAD

**Discussion by andrem-85** [pymupdf/PyMuPDF#2171] [Looking for help] [answered]
First of all, I love the pymupdf library. I'm a beginner but man I have been able to accomplish some great projects with it.

Anyway, I have a script where I am iterating over pages and inserting page numbers of a PDF package. I use shape.drawrect() and shape.insert_textbox() to put these page numbers in a red box.

Inside the loop I have:

```
            shape = doc.load_page(page).new_shape()
            shape.draw_rect(pages_rect)
            shape.finish(width = 1.05, color = red, dashes = "[] 0")
            rc = shape.insert_textbox(pages_rect, text, align=1, color = red, fontname = "hebo")
            shape.commit()
```

The code works very well but there is one issue: sometimes, seemingly randomly, the box draws with a dashed line once in an individual PDF package. This is why I have dashes = "[] 0" in shape.finish(). That did not seem to help.

Any ideas what's going on here?

Expected:
![image](https://user-images.githubusercontent.com/109046164/211926285-590da816-1e05-4041-8636-f928479465dc.png)

Unexpected:
![image](https://user-images.githubusercontent.com/109046164/211926176-4b7f995a-4e10-4a6d-9036-ba4982087bf5.png)

**Comment by JorjMcKie** [comment, marked as answer]
Thank you for the feedback ❤️!

Seems to be an existing page on which you are drawing?
Maybe previously existing drawing instructions come in your way, so try executing `page.clean_contents()` before you insert anything.
This does a lot of cleaning obviously, but most importantly encapsulates existing page appearance instructions by proper stack/unstack commands (`"q"`/`"Q"`).

**Reply by andrem-85** [reply]
That worked!! Thank you!

Yeah this script touches 100s of PDFs doing opening, merging, checking, modifying, etc. so one extra line of page.clean_contents() for this specific function is not a drama at all.

Thanks again :).
